### ISSUE TITLE
Project promotion

### ISSUE THREAD

**Issue by lukehinds** [sigstore/TSC#41]
Should projects promoted by sigstore be first subject to a promotion criteria (or should we even be promoting individual projects)?

As example, gitsign is on the landscape page and [openssf project page](https://openssf.org/community/sigstore/), got its own logo, but not policy-controller or a number of other tools (or any of the language specific libraries).

I recommend we explore this and perhaps implement some sort of promotion criteria. Either a project has reached maturity 1.0, has been leveraged in production x number of times and then voted by the TSC for inclusion etc. 

I guess another, perhaps meatier question is do we want sigstore to become a home of many projects. I know we already have a landscape, but do we want to become a CNCF style gig with an ecosystem? Or do we want to be software signing service that provides services and CLIs needed for said services.

**Comment by tracymiranda** [comment]
We did say we should do one for policy-controller - just no one got round to it.

**Comment by lukehinds** [comment]
policy-controller is just an example..I think the more key element is what should merit some projects having a presence (logos , blog update etc) over others. I know this has just played out organically as we have grown, and good intentions were at play. But i think it's worth having set expectations all round for the different projects (52 repos our present right now, although granted they are not all active / public facing).

**Comment by dlorenc** [comment]
> I think the more key element is what should merit some projects having a presence (logos , blog update etc) over others

I don't think this is a zero-sum game, and framing it as some projects getting a presence instead of others is probably detrimental. There are no "levels" or "tiers" in place today, the TSC accepts projects into Sigstore as a whole.

Could you clarify the concern around promotion? I don't actually understand the problem this is aiming to address. It would make it easier to understand the proposed solutions with a clearer description of the problems.

**Comment by lukehinds** [comment]
This is an explorative question, it's not meant to be detrimental to other projects, or at least it was not intended to be.

Right now we have four projects with their own logos (logos generally are used to promote a project, give them a brand), aka https://openssf.org/community/sigstore 

The projects are fulcio, rekor, cosign and gitsign. Three of those are what I would consider mature projects. They have reached 1.0. Have had security audit and well established, widely used.

We also have a CNCF style landscape, listing projects https://landscape.openssf.org/images/sigstore.pdf (with the same four)

Going forward, do we want any project that has a repo on here to have its own logo and placed onto a landscape?  Right now we have six libraries, helm-charts, three CLIs (cosign, gitsign, sget) a policy-controller, a kubectl plugin , three operators and lots more. 

Or do we want to have some criteria and tiers to determine quality, audit status etc? 

For the record, I am not singling out gitisign, I am fan and pretty sure I took to the TSC for approval. I would just like to have it clarified and seek the view of others.

**Comment by dlorenc** [comment]
I personally don't see any issues with the current status. The projects all self-declare their own maturity level using version numbers, and I don't think we have a shortage on getting logos so any of the projects that wants one should be able to get one. Rekor and Fulcio only went 1.0 a few weeks ago and had logos for a much longer period of time.

**Comment by SantiagoTorres** [comment]
I think a structure approach not necessarily to gatekeep but to better focus resources would be useful.  We have a lot of projects and helping them get due visibility is necessary from a strategic standpoint. 

As an aside: not sure exactly who made the call to what/who gets a logo,  but I would've assumed that having the same designer working on all of them is important

**Comment by dlorenc** [comment]
> I think a structure approach not necessarily to gatekeep but to better focus resources would be useful. We have a lot of projects and helping them get due visibility is necessary from a strategic standpoint.

Are there actually resource constraints though? IIUC all the existing work/resources are coming from the community so it's really not zero sum. I don't think the project is operating in a scarcity of resources that need to be allocated, more promotion of anything helps all the other projects. I don't see how keeping things off a landscape or telling projects they can't have logos will help here.

The gitsign logo was done here: https://github.com/sigstore/community/issues/67 I'm pretty sure any project that wants a logo can get one just by asking.

**Comment by davelester** [comment]
+1 to @lukehinds' suggestion, it’d be valuable to clarify org-wide Sigstore expectations about the project lifecycle in a way that's independent of the self-declared maturity level / version numbers of individual projects. That standard could also apply to libraries and critical parts of the ecosystem (language-specific clients, for example).

When a potential adopter or contributor comes to Sigstore, I’d like them to clearly see “here are the core projects within the Sigstore community that I can trust to have x,y,z” and “here’s the broader ecosystem” with an open list. 

I can imagine a lightweight process where the TSC votes to accept projects are top-level (TL) / core, after project criteria is clarified (I think @lukehinds has a good start with things like having a security audit, being well established, etc).

The related question of whether projects should have a sub-brand / logo may also be easier to answer once the project lifecycle is clarified. The current sub-branding scheme is not infinitely scalable (colors, marks, etc) – at the same time I love that we’re tying projects back to Sigstore itself.

**Comment by dlorenc** [comment]
I'm always very strongly opposed to adding process without clearly articulating the problems that the process is designed to address, and I still haven't seen that here. Process is always much harder to remove than it is to add, and it has a way of building up over time.

I'd strongly urge the TSC to not do anything here without a clear understanding of why they're doing it, including clear goals that the new process is required to meet.

**Comment by lukehinds** [comment]
Agree on @davelester points, it's the maturity / level of confidence we are communicating to users.

**Comment by trevrosen** [comment]
+1

I agree that an available delineation amongst projects is important to have available in our toolbox. I think GitSign is a good example of something that has promise and excitement, but that doesn't have the same level of strong buy-in and resource allocation as things like Rekor and Fulcio. How do we talk about such projects? What path do they walk when seeking further support? What signals can we offer beyond things like semver that don't capture maturity and adoption beyond relatively narrow scope? If I understand correctly, these are basically the questions that @lukehinds is trying to answer.

I *also* believe that such a process can't be instituted w/out very clear guidelines around what criteria enable graduation from one state to another, and that we need to be careful to avoid arbitrary gatekeeping.

As to "what problem is this solving", the answer to that (in my mind at least) is that I don't want people to come to Sigstore cold and see a flat landscape of projects and therefore make the (quite natural) assumption that they all enjoy the same level of support and buy-in when that's frequently just not the case. But today, that's an easy assumption for a n00b user to make. 😕

**Comment by dlorenc** [comment]
What beyond semver would it contain though? Do we have projects that self-declare themselves as stable and production-ready but we don't view them as such? 

If the problem is: "we should accurately explain the stability of projects and make that clear to users", that's one thing and I fully support it, but I think letting projects make that determination themselves with clear guidance and help is the right approach.

If the problem is: "we shouldn't put things on the landscape or let them have logos without going through a strict process", that's something different entirely.

**Comment by SantiagoTorres** [comment]
> The gitsign logo was done here: https://github.com/sigstore/community/issues/67 I'm pretty sure any project that wants a logo can get one just by asking.

I don't want to laser focus on this, but I'm relatively sure that other tools/projects asked for logos but it never happened, as opposed to gitsign. I don't see any sort of information on the issues that lets me see that gitsign did anything differently to other peojcts. I'm not sure why this turned out this way. Also, I don't believe there is foul play, if it needs to be said.

**Comment by dlorenc** [comment]
Which ones? Can they try filing an issue and seeing what happens?

**Comment by SantiagoTorres** [comment]
on sigstore/community#61 the phrasing was "are there any other subprojects that need a logo". I don't think it was clear it needed an issue.  I believe people asked for policy-controller, I brought it up for -monitoring. I remember sget being brought up in the community meeting.

Besides, I don't meant flaunt my tinfoil hat but sigstore/community#67 was opened at 4:01PM and three minutes later there was a logo to see and vote for, so I doubt it was being used as a tracking issue before requesting the logo.

**Comment by trevrosen** [comment]
@dlorenc where's the line between "clear guidance and help" and "a strict process"? What about when those giving guidance and those receiving it don't agree? How are such disputes to be adjudicated if the only criteria for getting the imprimatur of officialdom is having a repo in this org?

**Comment by dlorenc** [comment]
> on [sigstore/community#61](https://github.com/sigstore/community/issues/61) the phrasing was "are there any other subprojects that need a logo". I don't think it was clear it needed an issue. I believe people asked for policy-controller, I brought it up for -monitoring. I remember sget being brought up in the community meeting.
> 
> Besides, I don't meant flaunt my tinfoil hat but [sigstore/community#67](https://github.com/sigstore/community/issues/67) was opened at 4:01PM and three minutes later there was a logo to see and vote for, so I doubt it was being used as a tracking issue before requesting the logo.

I don't really know what to say here, if we want to delete the gitsign logo then go for it. If you want other projects to have logos, they're not gonna magically appear so someone will need to request it somehow. If someone wants a logo and didn't get one, we should fix that IMO.

**Comment by dlorenc** [comment]
> @dlorenc where's the line between "clear guidance and help" and "a strict process"? What about when those giving guidance and those receiving it don't agree? How are such disputes to be adjudicated if the only criteria for getting the imprimatur of officialdom is having a repo in this org?

That's one of the things the TSC is for, resolving cross-project conflicts. One approach here might be to say:

"Projects MUST receive a security audit before declaring a 1.0 release. To request one, follow this process."

Or:
"To request a logo, file an issue here in the repo." 

If for some reason there was a constraint on how many logos the project could receive and we were at the limit, you could even add:

"To request a logo, file an issue here in the repo. Please note that we'll prioritize requests for projects with 1.0 releases first."

**Comment by trevrosen** [comment]
@dlorenc OK I agree and so it sounds like we're... tacking toward a policy? 😅 

After all I think the crux of what @lukehinds is after here is simply having something written so that it can be known and referred to as opposed to dealt with one-off each time by the TSC. At that point we've got a rudimentary flow chart for TSC members and contributors to follow which minimizes hassle and surprise, which I think is progress! 😸

**Comment by lukehinds** [comment]
I think we might be getting off-track here with our focus on logos, and that is my fault. I was using the logos as a possible misconstrued signal of what we are communicating to users around stability of a project and perhaps we should have this better defined. The main consideration is how we communicate stability and projects being ready for use (even more so after the the recent GA announcement). 

It's not about stripping any project of its logo or giving them to all who want them, as @davelester said, a logo would likely be a byproduct of this.

**Comment by dlorenc** [comment]
> After all I think the crux of what @lukehinds is after here is simply having something written so that it can be known and referred to as opposed to dealt with one-off each time by the TSC. At that point we've got a rudimentary flow chart for TSC members and contributors to follow which minimizes hassle and surprise, which I think is progress! 😸

But what's the policy about? That's what I'm missing. Logos? 1.0? A landscape? 

I'm all for using a process to optimize stuff, but this hasn't exactly come up a lot.

**Comment by tracymiranda** [comment]
> on [sigstore/community#61](https://github.com/sigstore/community/issues/61) the phrasing was "are there any other subprojects that need a logo". I don't think it was clear it needed an issue. I believe people asked for policy-controller, I brought it up for -monitoring. I remember sget being brought up in the community meeting.
> 
> Besides, I don't meant flaunt my tinfoil hat but [sigstore/community#67](https://github.com/sigstore/community/issues/67) was opened at 4:01PM and three minutes later there was a logo to see and vote for, so I doubt it was being used as a tracking issue before requesting the logo.

Something I want to clear up on logos:
I worked to help get the new Sigstore logos done because I have experience with the LF processes here. It took forever, involved many iterations and endlessly chasing folks (especially with the changing PMs at OpenSSF). 
Then I also wanted to help refresh the other logos once Luke pointed out some flaws there and once again went through the process keeping the community informed on the way. Folks started to ask for all sorts of new logos in addition to the three we were doing. However the only one I was able to put a creative brief together for was gitsign, especially given the traction it was having on hackernews it was easier to explain what we were looking for from a user perspective. Plus I was keen to make sure this didn't have scope creep and end up taking forever. I'm really not sure why folks felt entitled to more logos without any follow up and @SantiagoTorres I don't appreciate the insinuations in your comments

**Comment by SantiagoTorres** [comment]
>I'm really not sure why folks felt entitled to more logos without any follow up and @SantiagoTorres I don't appreciate the insinuations in your comments

My insinuation is that it wasn't clear what the process was, and that the issue may not have been how the process was started (just by looking at the timelines). I said multiple times that I don't think there was foul play or ill intent whatsoever.

@tracymiranda I really appreciate all the hard work and expertise you put in keeping things running smoothly. Apologies if I came off as the opposite.

**Comment by tracymiranda** [comment]
Ok thanks for clarifying @SantiagoTorres - much appreciated.

**Comment by lukehinds** [comment]
I think this thread is becoming a little to heated now. I will make a PR (not sure where yet) as it looks like we seem to be agreeing that a process will be useful. We can then hammer out the specifics of said process in review.

**Comment by dlorenc** [comment]
> as it looks like we seem to be agreeing that a process will be useful.

For the record, I still disagree :) creating a process without a clear goal feels backwards.

**Comment by bobcallaway** [comment]
+1 to the TSC reviewing a proposal of *minimally-viable process* that helps external consumers and users of sigstore better understand project maturity, API stability, etc. I think there are some valuable perspectives above that should be considered while reviewing a proposal.

To be clear, I appreciate @tracymiranda's contributions here to help with the existing logos. IMO, the user experience of individual consumers, OSS developers, and package manager ecosystems should be one of our top community priorities.

**Comment by lukehinds** [comment]
How's this Dan?

> Define a process on how we communicate a project has reached a set maturity and is ready for general use?

**Comment by tracymiranda** [comment]
IMHO the Sigstore TSC should be looking at ways to grow the community in the form of contributors, projects, users etc. How do we get all of Rust projects to adopt Sigstore? Are our processes empowering new contributors?, Are we making it clear how to start a new subproject?etc
Coming up with reasons people *should not* use the subprojects wouldn't be high on my list of things to prioritize...

**Comment by davelester** [comment]
I'm viewing this effort as one that's promoting usage of projects rather than dissuading use.

I'd like to see greater inclusion of projects into the Sigstore website whether they are categorized as having reached a set of requirements or not -- and include them all! But when featuring additional projects, we'll want messaging that's crystal clear about each project's status so users know what to expect. I think this work will help towards that goal. 

Now that the project is GA and poised for future growth, it's a good time to clarify this -- I imagine it will become more challenging as the ecosystem grows. My hope is that we'll be able to help individual projects grow by including them in a broader ecosystem view that this enables.

**Comment by lukehinds** [comment]
> Coming up with reasons people should not use the subprojects wouldn't be high on my list of things to prioritize...

That really is not the intention. It's really not about telling people not to use projects, but instead communicating about what stage of majority the project is at. I was cack handed to use logos as an example and its thrown things off, but I did so as right now it seemed to be the only way we are expressing maturity and I figured we could be better communicating this for our end users. We did this very well with things like `COSIGN_EXPERIMENTAL` pre-GA and a few communities show their own maturity status such as [this](https://github.com/sigstore/sget) and [this](https://github.com/sigstore/sigstore-python), but we do not have a defined process for projects to help guide them. I believe its important to have one myself for our end users, especially if we intend to be a big tent and grow the eco-system out. I see this in much the same way as the attempts to define our architecture better, provide client specifications, all things that help out end users and projects too. 

I second others, you have done some wonderful work @tracymiranda , if anything we should have had things better defined for you from the onset.

**Comment by tracymiranda** [comment]
OK. My concerns stemmed from [the comment on November Roundup](https://docs.google.com/document/d/123WAHt4mxRbGN4dKiQCeYwACBES7jjtJr4QWcBF_e4o/edit?disco=AAAAjVztEFo) doc which seemed like it was suggesting we leave out talking about project updates if they weren't mature enough.

**Comment by lukehinds** [comment]
Not at all, I was simply referring to the conversation in this issue, and its another good example of the topic at hand.  Should we list every project that has had an update in the blog even if its just a minor release / with some bug fixes. Assume no ill-intent, I was the one who started the practise of listing updates, so I feel there is no harm in discussing these things openly.

**Comment by Hayden-IO** [comment]
Going to close this, as the community repo now tracks the requirements for project promotions.
